### PR TITLE
HIP-584: Fix error handling of reverts for all cases

### LINKED ISSUE #5620: HIP-584: Add encoded revert reason in JSON Error response
### Problem

We are not able to decode every type of revert reason into a human readable string, but still need to return the detailed revert reason.

### Solution

Add a new field called `data`, that is populated in the case of an error with the hexed revert reason returned from the EVM.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:

During relay tests for accepting hex revert reason from JSON error response two issues were identified:

1. Using `custom revert messages` causes unexpected exception in web3 module
2. Using `Panic(uint256)` reverts causes unexpected exception in web3 module

This PR fixes those issues and adds test coverage for all possible cases of revert mechanisms.

Repurpose `BytesDecoder` to convert only messages from `Error(string)`.

**Related issue(s)**:

Fixes #5620

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by georgi-l95 — DISMISSED** [review]
LGTM, this will unblock the relay for eth_call.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/convert/BytesDecoder.java [hunk 1]
```diff
@@ -32,26 +32,16 @@ public class BytesDecoder {
...
     private static final String ERROR_SIGNATURE = "0x08c379a0";
-    private static final Bytes ERROR_SIGNATURE_BYTES = Bytes.fromHexString(ERROR_SIGNATURE);
     private static final ABIType<Tuple> STRING_DECODER = TypeFactory.create("(string)");
     private static final int SIGNATURE_BYTES_LENGTH = 4;
 
-    public static String decodeEvmRevertReasonBytesToReadableMessage(final Bytes revertReason) {
-        if(revertReason == null || revertReason.isEmpty()) {
+    public static String maybeDecodeSolidityErrorStringToReadableMessage(final Bytes revertReason) {
+        if(revertReason == null || revertReason.isEmpty() || !revertReason.toHexString().startsWith(ERROR_SIGNATURE)
+                || revertReason.size() <= SIGNATURE_BYTES_LENGTH) {
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/convert/BytesDecoder.java:40]: Move this before the conversion to hex string as it's more performant to check it first.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/convert/BytesDecoder.java [hunk 2]
```diff
@@ -32,26 +32,16 @@ public class BytesDecoder {
...
-        }
-
-        final var tuple = STRING_DECODER.decode(encodedMessage.toArray());
-        if(tuple.size() > 0) {
-            return STRING_DECODER.decode(encodedMessage.toArray()).get(0);
-        } else {
-            return StringUtils.EMPTY;
-        }
+        final var encodedMessage = revertReason.slice(SIGNATURE_BYTES_LENGTH);
+        return STRING_DECODER.decode(encodedMessage.toArray()).get(0);
```
[the remark this answers is not in the document]
Comment (georgi-l95) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/convert/BytesDecoder.java]: Done.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/convert/BytesDecoder.java (hunk 1 above)
Comment (georgi-l95) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/convert/BytesDecoder.java]: Moved.

**Review by steven-sheehy — APPROVED** [review]

**Review by jnels124 — APPROVED** [review]
LGTM

**Review by mgoelswirlds — APPROVED** [review]

**Pull request merged**